### DISCUSSION TITLE
Sparkle does not check for updates on app launch

### DISCUSSION THREAD

**Discussion by leoshatrushin** [sparkle-project/Sparkle#2609] [Q&A] [answered]
### Description of the problem
I am following the Qt example and creating an SPUStandardUpdaterController with initWithStartingUpdater:YES, but nothing happens if I open the app twice, and nothing is printed to Console. If I use the "Check for updates..." menu bar action, it works.

### Do you use Sandboxing in your app?
No

### Version of `Sparkle.framework` in the latest version of your app
2.6.4

**Comment by zorgiepoo** [comment]
https://sparkle-project.org/documentation/#6-test-sparkle-out

Please first check:

* Run the app, then quit. By default, Sparkle doesn’t ask the user’s permission for checking updates until the second launch, in order to make your users’ first-launch impression cleaner.

* Run the app again. The update process should proceed as expected. **Note by default, Sparkle checks for updates in the background once every 24 hours**. To test automatic update checks immediately, run defaults delete my-bundle-id SULastCheckTime to clear the last update check time before launching the app. Alternatively, initiate a manual update check from the app’s menu bar.

**Comment by leoshatrushin** [comment]
The app never asks for permission. The defaults domain is not created or updated until I do a manual update check. When I first run the app, the return value of [_host objectForUserDefaultsKey:SUEnableAutomaticChecksKey] in
```
// If the user has been asked about automatic checks, don't bother prompting
    // When the user answers to the permission prompt, this will be set to either @YES or @NO instead of nil
    if ([_host objectForUserDefaultsKey:SUEnableAutomaticChecksKey] != nil) {
        shouldPrompt = NO;
    }
 ```
In SPUUpdater.m:385 is
```
rax = 0x00007ff859cf7718  CoreFoundation`__kCFBooleanFalse
```
Not 'nil' as suggested, even though my Info.plist does not set a value for SUEnableAutomaticChecks, nor SUAutomaticallyUpdate

**Reply by zorgiepoo** [reply]
On the first launch, Sparkle should set `SUHasLaunchedBefore` in your defaults domain (Sparkle always ensures this gets set when starting the updater)

On the second launch, Sparkle should prompt a permission dialog since `SUHasLaunchedBefore` was set, and `SUEnableAutomaticChecks` is not specified in the Info.plist, and the user hasn't already responded to a permission request (hence populating `SUEnableAutomaticChecks` in your user defaults), and the app hasn't programmatically changed this state before (https://sparkle-project.github.io/documentation/api-reference/Classes/SPUUpdater.html#/c:objc(cs)SPUUpdater(py)automaticallyChecksForUpdates -- also updates `SUEnableAutomaticChecks` in user defaults).

Make sure you use `defaults read your-app-bundle-id` to inspect the user defaults. Don't inspect the files on disk as those may not be updated/flushed. You can `defaults delete your-app-bundle-id` to delete your app's defaults to a clean state if necessary. If your app was previously sandboxed and is not anymore, that is also been known to confuse the user defaults state.

If you're using ARC also make sure your `SPUStandardUpdaterController` instance is held on for lifetime of your app.

If you still can't figure it out you can attach a sample code, or a sample app binary or what not, etc, that can make the issue reproducible to me. Note the Sparkle Test App works as expected.

**Reply by leoshatrushin** [reply, marked as answer]
`defaults delete your-app-bundle-id ` to delete the app defaults to a clean state worked. Thanks
